File: app/src/main/java/com/zzl/top/common/http/impl/BaseResponse.java
package com.zzl.top.common.http.impl;

import com.zzl.top.common.http.IResponse;

/**
 * Created by zhangzl
 * time: 2017/10/16 20:48.
 * 描述：
 */

public class BaseResponse implements IResponse {

    public static final int STATE_UNKNOWN_ERROR = 100001;
    public static final int STATUS_OK = 200;

    /**
     * 状态码
     */
    private int code;

    /**
     * 响应数据
     */
    private String data;

    @Override
    public String getData() {
        return data;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public void setData(String data) {
        this.data = data;
    }
}

File: http/src/main/kotlin/com/zzl/net/impl/BaseResponse.kt
package com.zzl.net.impl

import com.zzl.net.IResponse

/**
 * Created by zhangzl
 * 时间： 2017/10/29 14:27
 * 描述：
 */
open class BaseResponse : IResponse {

    /**
     * 状态码
     */
    private var code = 0

    /**
     * 响应数据
     */
    private var data: String? = null

    override fun getCode(): Int {
        return code
    }

    override fun getData(): String {
        return data ?: "{}"
    }

    override fun setCode(code: Int) {
        this.code = code
    }

    override fun setData(data: String?) {
        this.data = data
    }
}